slatedb/src/db.rs

```diff
@@ -3144,6 +3147,70 @@ mod tests {
         assert_eq!(db.inner.mono_clock.last_tick.load(Ordering::SeqCst), 11);
     }
 
+    #[tokio::test]
+    async fn test_put_get_reopen_delete_with_separate_wal_store() {
+        let object_store = Arc::new(InMemory::new());
+        let wal_object_store = Arc::new(InMemory::new());
+        let kv_store = Db::builder("/tmp/test_kv_store", object_store.clone())
+            .with_settings(test_db_options(0, 1024, None))
+            .with_wal_object_store(wal_object_store.clone())
+            .build()
+            .await
+            .unwrap();
+
+        let key = b"test_key";
+        let value = b"test_value";
+        kv_store.put(key, value).await.unwrap();
+        kv_store.flush().await.unwrap();
```

Comment:
Could you set a `fail_point` in memtable flush and validate that the `wal_object_store` contains the SST? As the test stands, it doesn't actually seem to validate that the data hits the wal object storage.

Reply:
you mean to validate for SSTs inside the fail point? or to validate in the test body itself, what's the purpose of the fail point then? could you provide some existing test doing this kind of a validation as an example?

Reply:
I think the test should:

1. Pause memtable flushes.
2. Call put
3. Call flush
4. Validate that the data is in the underlying WAL object store using a list/get.
5. Unpause memtable flushes.
6. Call flush again.
7. Validate the data is in the main object store using list/get.

To pause memtable flushes, you can use:

```rust
// pause
fail_parallel::cfg(fp_registry.clone(), "write-compacted-sst-io-error", "pause").unwrap();
// ...
// unpause
fail_parallel::cfg(fp_registry.clone(), "write-compacted-sst-io-error", "off").unwrap();
```